src/amp-story-player/amp-story-player-impl.js

```diff
@@ -486,6 +487,39 @@ export class AmpStoryPlayer {
     }
   }
 
+  /**
+   * Navigates stories given a number.
+   * @param {number} storyDelta
+   */
+  go(storyDelta) {
+    if (this.currentIdx_ + storyDelta >= this.stories_.length || this.currentIdx_ + storyDelta < 0) {
+      throw new Error('Out of Story range.')
+    }
+    
+    if (storyDelta === 0) {
+      return;
+    }
+
+    const newIndex = this.currentIdx_ + storyDelta;
+
+    const previousStory = this.stories_[newIndex - 1];
+    this.updatePreviousIframe_(
+      previousStory[IFRAME_IDX],
+      IframePosition.PREVIOUS
+    );
+
+    const currentStory = this.stories_[newIndex];
+    this.updateCurrentIframe_(currentStory[IFRAME_IDX]);
+
+    const nextStoryIdx = newIndex + 1;
+    if (
+      nextStoryIdx < this.stories_.length &&
+      this.stories_[nextStoryIdx][IFRAME_IDX] === undefined
+    ) {
+      this.allocateIframeForStory_(nextStoryIdx);
```

Comment:
I'm not sure this will work as it is. The way this function is implemented now, it will deallocate an iframe from the story furthest away and rotate the iframe pool with the assumption that we are moving in a linear fashion and not skipping stories. I think if we test all the edge cases we will find something that breaks.

IMO we have two options:

1) use the show method here so that it can take care of this
2) update the `allocateIframeForStory_` method to take into account this edge case.

Reply:
Updated to use the show method!